src/System.Data.SqlClient/tests/ManualTests/SQL/MirroringTest/ConnectionOnMirroringTest.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public static class ConnectionOnMirroringTest
+    {
+        private static ManualResetEvent mre = new ManualResetEvent(false);
+
+        [CheckConnStrSetupFact]
+        public static void TestMultipleConnectionToMirroredServer()
+        {
+            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(DataTestUtility.TcpConnStr);
+            builder.ConnectTimeout = 0;
+            string connectionString = builder.ConnectionString;
+
+            string mirroringStateDesc;
+            string failoverPartnerName;
+            bool isMirroring = GetMirroringInfo(connectionString, out mirroringStateDesc, out failoverPartnerName);
+            bool isSynchronized = "SYNCHRONIZED".Equals(mirroringStateDesc, StringComparison.InvariantCultureIgnoreCase);
+            List<SqlConnection> list = new List<SqlConnection>();
+            if (isMirroring && isSynchronized && !string.IsNullOrEmpty(failoverPartnerName))
+            {
+                TestWorker worker = new TestWorker(connectionString);
+                Thread childThread = new Thread(() => worker.TestMultipleConnection());
+                childThread.Start();
+
+                mre.Reset();
```

Comment:
Don't need to call reset, since the initial state is set to nonsignaled by passing false to the constructor (as you've done above).